internal/db/service/user_history.go

```diff
@@ -0,0 +1,137 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	snitchv1 "snitch/pkg/proto/gen/snitch/v1"
+
+	"connectrpc.com/connect"
+)
+
+// CreateUserHistory creates a new user history entry in the group database
+func (s *DatabaseService) CreateUserHistory(
+	ctx context.Context,
+	req *connect.Request[snitchv1.DbCreateUserHistoryRequest],
+) (*connect.Response[snitchv1.DbCreateUserHistoryResponse], error) {
+	db, err := s.getOrCreateGroupDB(ctx, req.Msg.GroupId)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to get group database: %w", err))
+	}
+
+	// Ensure user and server exist
+	if err := s.ensureUserExists(ctx, db, req.Msg.UserId); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to ensure user exists: %w", err))
+	}
+	if err := s.ensureServerExists(ctx, db, req.Msg.ServerId); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to ensure server exists: %w", err))
+	}
+
+	query := `INSERT INTO user_history (user_id, server_id, action, reason, evidence_url) 
```

Comment:
Okay I got this far into the PR before feeling like commenting what happened to sqlc. It was nice, we could still use it in the db service